supersetapiclient/base.py

```diff
@@ -21,6 +21,18 @@ def default_string():
     return dataclasses.field(default="", repr=False)
 
 
+def raise_for_status(response):
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        # Attempt to propagate the server error message
+        try:
+            error_msg = response.json()["message"]
+        except Exception:
+            raise e
+        raise ServerError(*e.args, request=e.request, response=e.response, message=error_msg) from None
+
+
```

Comment:
I was trying to avoid diving into the logic of `response.raise_for_status()` and speculate about which `status_code` values trigger an error. I think catching `HTTPError` should be OK because we only re-raise it as something else if `response.json()["message"]` succeeds. Otherwise, it is passed unaltered.

Reply:
I'm expanding the test suite and encountering other HTTP 4xx errors. So I think the above approach with less assumptions about the original exception instance is better.